src/Animations/babylon.animatable.ts

```diff
@@ -282,21 +282,26 @@
         /**
          * Stop and delete the current animation
          * @param animationName defines a string used to only stop some of the runtime animations instead of all
+         * @param targetMask - a function that determines if the animation should be stopped based on its target (all animations will be stopped if both this and animationName are empty)
          */
-        public stop(animationName?: string): void {
-            if (animationName) {
+        public stop(animationName?: string, targetMask?: (target: any) => boolean): void {
+            if (animationName || targetMask) {
                 var idx = this._scene._activeAnimatables.indexOf(this);
 
                 if (idx > -1) {
 
                     var runtimeAnimations = this._runtimeAnimations;
 
                     for (var index = runtimeAnimations.length - 1; index >= 0; index--) {
-                        if (typeof animationName === "string" && runtimeAnimations[index].animation.name != animationName) {
+                        const runtimeAnimation = runtimeAnimations[index];
+                        if (typeof animationName === "string" && runtimeAnimation.animation.name != animationName) {
+                            continue;
+                        }
+                        if (typeof targetMask === "function" && !targetMask(runtimeAnimation.target)) {

```

Comment:
no need to test the type here, we assume it is a function if targetMask exists

Reply:
I was just pattern-matching the lines for animation name. Was that type checking just to prevent undefined matching on both sides?

Reply:
Well both can be removed actually and replaced by if (xx) :)